### PR TITLE
Batched syr

### PR DESCRIPTION
**Pull request by TorreZuk**
adds batched and strided_batched syr level 2
all tests in syr_gtest.yaml

Summary of testing:
924 tests from quick/syr (1424 ms total)
74 tests from pre_checkin/syr (9101 ms total)
12 tests from nightly/syr (13732 ms total)
1848 tests from quick/syr_batched (1506 ms total)
146 tests from pre_checkin/syr_batched (36002 ms total)
24 tests from nightly/syr_batched (54787 ms total)
1848 tests from quick/syr_strided_batched (1479 ms total)
146 tests from pre_checkin/syr_strided_batched (33531 ms total)
24 tests from nightly/syr_strided_batched (52491 ms total)
5046 tests from 9 test cases ran. (204061 ms total)

### CODE REVIEW

File: library/include/rocblas-functions.h [hunk 1]
```diff
@@ -2168,6 +2168,122 @@ rocblas_zsyr(rocblas_handle handle,
...
 */
 
+/*! \brief BLAS Level 2 API
+
+    \details
+    xSYR_batched performs a batch of matrix-vector operations
+
+        A[i] := A[i] + alpha*x[i]*x[i]**T
+
+    where alpha is a scalars, x is an array of vectors, and A is an array of
```
Comment (daineAMD) [inline library/include/rocblas-functions.h]: typo: alpha is a scalar

File: library/include/rocblas-functions.h [hunk 2]
```diff
@@ -2168,6 +2168,122 @@ rocblas_zsyr(rocblas_handle handle,
...
+
+    ********************************************************************/
+
+ROCBLAS_EXPORT rocblas_status rocblas_ssyr_batched(rocblas_handle     handle,
+                                                   rocblas_fill       uplo,
+                                                   rocblas_int        n,
+                                                   const float*       alpha,
+                                                   const float* const x[],
+                                                   rocblas_int        incx,
+                                                   float*             A[],
```
Comment (daineAMD) [inline library/include/rocblas-functions.h]: Can this be a float* const A[] ?

File: library/include/rocblas-functions.h (hunk 1 above)
Comment (TorreZuk) [reply library/include/rocblas-functions.h]: fixed for syr and ger

File: library/include/rocblas-functions.h (hunk 2 above)
Comment (TorreZuk) [reply library/include/rocblas-functions.h]: made the ptrs const

File: clients/gtest/syr_gtest.cpp [hunk 3]
```diff
@@ -7,14 +7,80 @@
...
+            name << rocblas_datatype2string(arg.a_type);
+
+            if(strstr(arg.function, "_bad_arg") != nullptr)
+            {
+                name << "_bad_arg";
+            }
+            else
+            {
+                name << '_' << (char)std::toupper(arg.uplo) << '_' << arg.N << '_' << arg.alpha
+                     << '_' << arg.incx << '_' << arg.lda;
```
Comment (mahmoodw) [inline clients/gtest/syr_gtest.cpp]: I think we would want to keep the order of parameters in the name consistent with the argument list in the API for better readability when debugging. i.e.  incx << stride_x << lda << strideA << batch_count

File: library/include/rocblas-functions.h [hunk 4]
```diff
@@ -2417,6 +2417,122 @@ rocblas_zsyr(rocblas_handle handle,
...
+              specifies the increment for the elements of x_i.
+    @param[in]
+    stridex      specifies the pointer increment between vectors (x_i) and (x_i+1).
+    @param[inout]
+    A         pointer to the first matrix A on the GPU.
+    @param[in]
+    lda       rocblas_int
+              specifies the leading dimension of A.
+    @param[in]
+    strideA      stride from the start of one matrix (A_i) and the next one (A_i+1)
```
Comment (mahmoodw) [inline library/include/rocblas-functions.h]: Looks like we should be specifying that rocblas_stride is used for stride parameters

File: library/src/blas2/rocblas_syr.hpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ * ************************************************************************ */
+#pragma once
+#include "handle.h"
+#include "rocblas.h"
+
+template <typename T, typename U>
+__global__ void rocblas_syr_kernel(rocblas_fill uplo,
+                                   rocblas_int  n,
+                                   U            alpha_device_host,
+                                   const T* __restrict__ x,
```
Comment (mahmoodw) [inline library/src/blas2/rocblas_syr.hpp]: Might want to ask Juan if he needs support in the template for a vector of the scalar alpha. We may just want to stay consistent and it anyway. You can see rocblas_scal as an example, just need to add in a stride for alpha and call the new load_scalar helper.

File: library/include/rocblas-functions.h (hunk 4 above)
Comment (TorreZuk) [reply library/include/rocblas-functions.h]: Yes, maybe a separate PR and update all strided functions as I see others that don't indicate the type for strides?   some like GER strided are still rocblas_int but should we update to rocblas_stride now.   Other parameters in doxygen don't indicate the type either and we should probably make it all consistent.    I'll do syr now.

File: library/src/blas2/rocblas_syr.hpp (hunk 5 above)
Comment (TorreZuk) [reply library/src/blas2/rocblas_syr.hpp]: I'll ask Juan.

File: clients/gtest/syr_gtest.cpp (hunk 3 above)
Comment (TorreZuk) [reply clients/gtest/syr_gtest.cpp]: Sure thing I'll make it consistent to others, but then comparing non batched to batched then isn't easy via filter.

File: library/include/rocblas-functions.h (hunk 4 above)
Comment (mahmoodw) [reply library/include/rocblas-functions.h]: I can do that as part of my refactor

File: clients/include/testing_syr_batched.hpp [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        rocblas_status_invalid_handle);
+}
+
+template <typename T>
+void testing_syr_batched(const Arguments& arg)
+{
+    rocblas_int  N           = arg.N;
+    rocblas_int  incx        = arg.incx;
+    rocblas_int  lda         = arg.lda;
+    T            h_alpha     = (T)arg.alpha;
```
[the remark this answers is not in the document]
Comment (TorreZuk) [reply clients/include/testing_syr_batched.hpp]: Had just copied ger approach, which I see has changed.

File: library/src/blas2/rocblas_syr.hpp (hunk 5 above)
Comment (TorreZuk) [reply library/src/blas2/rocblas_syr.hpp]: I'll add the alpha vector support following the other examples pattern.

Comment (TorreZuk) [reply library/src/blas2/rocblas_syr.hpp]: @mahmoodw  please review again.  Added vector but deferred supporting host vector.

**Pull request merged**